src/App.tsx

```diff
@@ -1,61 +1,118 @@
 import './App.scss';
-
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import usersFromServer from './api/users';
+import { User } from './types/Types';
+import { Todo } from './types/Types';
+import todosFromServer from './api/todos';
+import React, { useState } from 'react';
 
 export const App = () => {
+  const initialTodos: Todo[] = todosFromServer.map(todo => ({
+    ...todo,
+    user: usersFromServer.find(u => u.id === todo.userId) as User,
+  }));
+
+  const [todos, setTodos] = useState<Todo[]>(initialTodos);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState<number>(0);
+  const [errors, setErrors] = useState({
+    title: '',
+    user: '',
+  });
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+
+    let hasErrors = false;
+    const newErrors = { title: '', user: '' };
+
+    if (!title.trim()) {
+      newErrors.title = 'Please enter a title';
+      hasErrors = true;
+    }
+
+    if (userId === 0) {
+      newErrors.user = 'Please choose a user';
+      hasErrors = true;
+    }
+
+    setErrors(newErrors);
+
+    if (hasErrors) {
+      return;
+    }
+
+    const newTodo: Todo = {
+      id: todos.length > 0 ? Math.max(...todos.map(todo => todo.id)) + 1 : 1,
+      title: title.trim(),
+      userId,
+      completed: false,
+      user: usersFromServer.find(u => u.id === userId) as User,
```

Comment:
Similar to line 12, using `as User` assumes that a user will always be found. Consider adding a check to ensure that `find` does not return `undefined` to avoid potential runtime errors.